### PR TITLE
feat: add signup page

### PR DESCRIPTION
**Pull request by ini-s**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  // const [user, setUser] = useState(null)
+  if (!user) {
+    return <SignIn />
+  }
+
+  return (
+    <div className="App">
+      <Header />
+      <main>
+        <HomePage />
```
Comment (Asylcreek) [inline src/App.tsx]: This should not be here

File: src/index.tsx [hunk 2]
```diff
@@ -2,11 +2,12 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
+import { BrowserRouter } from 'react-router-dom';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
```
Comment (Asylcreek) [inline src/index.tsx]: Wrap `BrowserRouter` with `React.StrictMode`

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (ini-s) [reply src/App.tsx]: What should not be there?

**Comment by ini-s** [comment]
The  SignIn page is still not rendering

File: src/App.tsx [hunk 3]
```diff
@@ -1,28 +1,24 @@
...
-        <HomePage />
-
         <Routes>
-          <Route path="/" element={<HomePage />} />
-          <Route path="/signin" element={<SignIn />} />
+          {user ?
+            <Route path="/" element={<HomePage />} /> :
+            <Route path="/signin" element={<SignIn />} />
+          }
           <Route path="/signup" element={<SignUp />} />
```
Comment (Asylcreek) [inline src/App.tsx]: This is incomplete

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (ini-s) [reply src/App.tsx]: I haven't been able to find what's missing

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**